modules/terraform/workspace_test.go

```diff
@@ -130,3 +131,99 @@ func TestNameMatchesWorkspace(t *testing.T) {
 		assert.Equal(t, testCase.expected, actual, "Name: %q, Workspace: %q", testCase.name, testCase.workspace)
 	}
 }
+
+func TestWorkspaceDeleteE(t *testing.T) {
+	t.Parallel()
+
+	// state describes an expected status when a given testCase begins
+	type state struct {
+		workspaces []string
+		current    string
+	}
+
+	// testCase describes a named test case with a state, args and expcted results
+	type testCase struct {
+		name              string
+		initialState      state
+		toDeleteWorkspace string
+		expectedCurrent   string
+		expectedError     error
+	}
+
+	testCases := []testCase{
+		{
+			name: "delete another existing workspace and stay on current",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace: "production",
+			expectedCurrent:   "staging",
+			expectedError:     nil,
+		},
+		{
+			name: "delete current workspace and switch to a specified",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "production",
+			},
+			toDeleteWorkspace: "production",
+			expectedCurrent:   "default",
+			expectedError:     nil,
+		},
+		{
+			name: "delete a non existing workspace should trigger an error",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace: "hellothere",
+			expectedCurrent:   "staging",
+			expectedError:     WorkspaceDoesNotExist("hellothere"),
+		},
+		{
+			name: "delete the default workspace triggers an error",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace: "default",
+			expectedCurrent:   "staging",
+			expectedError:     &UnsupportedDefaultWorkspaceDeletion{},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Add `t.Parallel()` here.

Reply:
Wow, I'm impressed by the performance induced by adding this option (went to ~8s to ~2s on my machine).